clash-prelude/src/Clash/Xilinx/ClockGen.hs

```diff
@@ -95,11 +106,8 @@ clockWizardDifferential
   -- ^ Free running clock, positive phase
   -> Reset domIn
   -- ^ Reset for the PLL
-  -> (Clock domOut, Enable domOut)
+  -> (Clock domOut, Signal domPllLock Bool)
   -- ^ (Stable PLL clock, PLL lock)
-clockWizardDifferential !_name (Clock _ Nothing) (Clock _ Nothing) rst =
-  (Clock SSymbol Nothing, unsafeCoerce (toEnable (unsafeToHighPolarity rst)))
-clockWizardDifferential !_name _ _ _ =
-  error "clockWizardDifferential: no support for dynamic clocks"
+clockWizardDifferential !_ !_ = clocks
```

Comment:
Hah, I'm asking you to do A, you do A, and now I'm asking you to do B. Because I later realised this solution has a disadvantage: if people get creative with different clocks connected to the two clock inputs here, they will only encounter the `clocks: dynamic clocks unsupported` error message when they do it wrong on the positive phase. The negative phase is no longer checked. So could you change it to:

```suggestion
clockWizardDifferential !_ clk@(Clock _ Nothing) (Clock _ Nothing) =
  clocks clk
clockWizardDifferential !_ _ _ =
  error "clockWizardDifferential: dynamic clocks not supported"
```

(the rephrasing of the error message is intentional, I'm trying to go one step towards a single phrasing in `clash-prelude`).